js/app/src/init.ts

```diff
@@ -160,6 +160,9 @@ export function create_components(): {
 			instance.props.server_fns,
 			app
 		);
+		// Pass the max file size from the app to the component
+		// so we can validate it before files are uploaded
+		instance.props.max_file_size = app.config.max_file_size;
```

Comment:
Rather than adding this here, i think it might make sense to add it to the `Gradio` object that gets passed into each component. The idea of of that object is that it contains app level configuration (ie globals) where as props should just be specific to the component.

We should be able to pass the value through to the `Render.svelte` via blocks and then add it as an additional arg to `new Gradio()`